dask_expr/_shuffle.py

```diff
@@ -91,6 +93,18 @@ def __str__(self):
     def _node_label_args(self):
         return [self.frame, self.partitioning_index]
 
+    @functools.cached_property
+    def _partitioning_index(self):
+        partitioning_index = self.partitioning_index
+        if isinstance(partitioning_index, (str, int)):
```

Comment:
I assume this is a check for whether this index is a column?
```suggestion
        if partitioning_index in self.frame.columns:
```

Reply:
partitioning_index could be an expression, and in with expressions doesn't work the way we need it here